src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/CachePoolClear/config.yml

```diff
@@ -0,0 +1,16 @@
+imports:
+    - { resource: ../config/default.yml }
+
+services:
+    dummy1:
+        class: Symfony\Bundle\FrameworkBundle\Tests\Fixtures\DeclaredClass
+        arguments: ['@cache.private_pool']
+    dummy2:
+        parent: dummy1
```

Comment:
In fact this is not a bug, it's consistent with the fact that you can't clear a private service right now.
Here is the patch that you can apply to fix this:
```diff
diff --git a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php
index 00dfdee..20e3da6 100644
--- a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php
+++ b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolClearerPass.php
@@ -34,7 +34,10 @@ final class CachePoolClearerPass implements CompilerPassInterface
                 if (isset($attr['clearer'])) {
                     $poolsByClearer[$attr['clearer']][$id] = new Reference($id);
                 }
-                if (array_key_exists('clearer', $attr)) {
+                if (!empty($attr['unlazy'])) {
+                    $container->getDefinition($id)->setLazy(false);
+                }
+                if (array_key_exists('clearer', $attr) || array_key_exists('unlazy', $attr)) {
                     break;
                 }
             }
diff --git a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php
index 593b778..4c12a95 100644
--- a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php
+++ b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/CachePoolPass.php
@@ -47,8 +47,10 @@ class CachePoolPass implements CompilerPassInterface
             if ($pool->isAbstract()) {
                 continue;
             }
+            $isLazy = $pool->isLazy();
             while ($adapter instanceof DefinitionDecorator) {
                 $adapter = $container->findDefinition($adapter->getParent());
+                $isLazy = $isLazy || $adapter->isLazy();
                 if ($t = $adapter->getTag('cache.pool')) {
                     $tags[0] += $t[0];
                 }
@@ -80,8 +82,16 @@ class CachePoolPass implements CompilerPassInterface
                 throw new InvalidArgumentException(sprintf('Invalid "cache.pool" tag for service "%s": accepted attributes are "clearer", "provider", "namespace" and "default_lifetime", found "%s".', $id, implode('", "', array_keys($tags[0]))));
             }
 
+            $attr = array();
             if (null !== $clearer) {
-                $pool->addTag('cache.pool', array('clearer' => $clearer));
+                $attr['clearer'] = $clearer;
+            }
+            if (!$isLazy) {
+                $pool->setLazy(true);
+                $attr['unlazy'] = true;
+            }
+            if ($attr) {
+                $pool->addTag('cache.pool', $attr);
             }
         }
     }
diff --git a/src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/CachePoolClear/config.yml b/src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/CachePoolClear/config.yml
index c96183e..393c554 100644
--- a/src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/CachePoolClear/config.yml
+++ b/src/Symfony/Bundle/FrameworkBundle/Tests/Functional/app/CachePoolClear/config.yml
@@ -2,11 +2,9 @@ imports:
     - { resource: ../config/default.yml }
 
 services:
-    dummy1:
+    dummy:
         class: Symfony\Bundle\FrameworkBundle\Tests\Fixtures\DeclaredClass
         arguments: ['@cache.private_pool']
-    dummy2:
-        parent: dummy1
 
 framework:
     cache:
```